src/ort_genai_c.h

```diff
@@ -76,6 +80,22 @@ OGA_EXPORT OgaResult* OGA_API_CALL OgaGenerator_GenerateNextToken_TopP(OgaGenera
  */
 OGA_EXPORT OgaResult* OGA_API_CALL OgaGenerator_GetSequence(const OgaGenerator*, int index, int32_t* tokens, size_t* count);
 
+OGA_EXPORT OgaResult* OGA_API_CALL OgaCreateTokenizer(const OgaModel* model, OgaTokenizer** out);
+OGA_EXPORT void OGA_API_CALL OgaDestroyTokenizer(OgaTokenizer*);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaTokenizerEncode(const OgaTokenizer*, const char* const* strings, size_t count, OgaSequences** out);
+
+/* Decode a single token sequence and returns a null terminated utf8 string. out_string must be freed with OgaDestroyString
+ */
+OGA_EXPORT OgaResult* OGA_API_CALL OgaTokenizerDecode(const OgaTokenizer*, const int32_t* tokens, size_t token_count, const char** out_string);
+
+OGA_EXPORT OgaResult* OGA_API_CALL OgaCreateTokenizerStream(const OgaTokenizer*, OgaTokenizerStream** out);
+OGA_EXPORT void OGA_API_CALL OgaDestroyTokenizerStream(OgaTokenizerStream*);
+
+/* Streaming decoder for tokens. Allows displaying a decoded token stream as the tokens are generated.
+ * As not every token will result in bytes written to the string, bytes_written may sometimes be zero.
+ */
+OGA_EXPORT OgaResult* OGA_API_CALL OgaTokenizerDecodeStream(const OgaTokenizer*, OgaTokenizerStream*, int32_t token, char* string, size_t string_size, size_t* bytes_written);
```

Comment:
>int32_t token [](http://example.com/codeflow?start=102&length=13)

we need to support multiple batch

Reply:
If the user needs to decode multiple streams, the user should iterate over multiple OgaTokenizerStream objects.

Otherwise we'd need to create more C API types, OgaTokenizerStreams, and have it return an array of strings each iteration. Is that worth it? There's no perf gain.

Reply:
@RyanUnderhill please add sample usage to the PR

Reply:
Yep, there's a sample of all the APIs added to c_api.cpp: TokenizerCAPI